src/AbstractSQLOptimiser.ts

```diff
@@ -627,8 +627,47 @@ const typeRules: Dictionary<MatchFn> = {
 		}),
 		matchArgs('IsDistinctFrom', AnyValue, AnyValue),
 	),
-	Add: MathOp('Add'),
-	Subtract: MathOp('Subtract'),
+	Add: tryMatches(
+		MathOp('Add'),
+		// redirecting for non both numeric values to try the date operator matcher
+		// if more arithmetic operation operands than numeric and date / interval are needed. An additional matcher / redirects are needs to be used in the tryMatcher chain
+		Helper<OptimisationMatchFn>((args) => {
+			// checkargs not necessary as checked in the following matchArgs of date operator matcher
+			return ['AddDate', ...args];
+		}),
+	),
+	Subtract: tryMatches(
+		MathOp('Subtract'),
+		// redirecting for non both numeric values to try the date operator matcher
+		// if more arithmetic operation operands than numeric and date / interval are needed. An additional matcher / redirects are needs to be used in the tryMatcher chain
+		Helper<OptimisationMatchFn>((args) => {
+			// checkargs not necessary as checked in the following matchArgs of date operator matcher
+			return ['SubtractDate', ...args];
+		}),
+	),
+	SubtractDate: tryMatches(
+		// checks every matcher and will proceed with the first one satisfying the TypeMatch
+		// Fall through helper matcher to throw an error when the operand types don't match.
+		matchArgs('SubtractDate', DateValue, DateValue),
+		matchArgs('SubtractDate', DateValue, DurationValue),
+		matchArgs('SubtractDate', DateValue, NumericValue),
+		Helper<OptimisationMatchFn>((args) => {
+			throw new SyntaxError(
+				`"SubtractDate" expects (DateValue, DateValue | DurationValue | NumericValue) but given ${args}`,
+			);
+		}),
+	),
+	AddDate: tryMatches(
+		// checks every matcher and will proceed with the first one satisfying the TypeMatch
+		// Fall through helper matcher to throw an error when the operand types don't match.
+		matchArgs('AddDate', DateValue, DurationValue),
+		matchArgs('AddDate', DateValue, NumericValue),
+		Helper<OptimisationMatchFn>((args) => {
+			throw new SyntaxError(
+				`"AddDate" expects (DateValue, DurationValue | NumericValue) but given ${args}`,
+			);
+		}),
```

Comment:
The helper wrapper should be unnecessary?
```suggestion
		(args) => {
			throw new SyntaxError(
				`"AddDate" expects (DateValue, DurationValue | NumericValue) but given ${args}`,
			);
		},
```